pkg/controllers/nodeclass/validation.go

```diff
@@ -40,6 +60,125 @@ func (n Validation) Reconcile(ctx context.Context, nodeClass *v1.EC2NodeClass) (
 			fmt.Sprintf("%q tag does not pass tag validation requirements", offendingTag))
 		return reconcile.Result{}, reconcile.TerminalError(fmt.Errorf("%q tag does not pass tag validation requirements", offendingTag))
 	}
+	//Auth Validation
+	nodeClaim := &karpv1.NodeClaim{
+		Spec: karpv1.NodeClaimSpec{
+			NodeClassRef: &karpv1.NodeClassReference{
+				Name: nodeClass.ObjectMeta.Name,
+			},
+		},
+	}
+	tags, err := utils.GetTags(nodeClass, nodeClaim, options.FromContext(ctx).ClusterName)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get Tags, %w", err)
+	}
+
+	createFleetInput := instance.GetCreateFleetInput(nodeClass, string(karpv1.CapacityTypeOnDemand), tags, mockLaunchTemplateConfig())
+	createFleetInput.DryRun = aws.Bool(true)
+
+	if _, err := n.ec2api.CreateFleet(ctx, createFleetInput); awserrors.IgnoreDryRunError(err) != nil {
+		nodeClass.StatusConditions().SetFalse(v1.ConditionTypeValidationSucceeded, "CreateFleetAuthCheckFailed", fmt.Sprintf("unauthorized operation %s", err))
+		return reconcile.Result{}, reconcile.TerminalError(fmt.Errorf("create fleet %w", err))
+	}
+
+	createLaunchTemplateInput := launchtemplate.GetCreateLaunchTemplateInput(mockOptions(*nodeClaim, nodeClass, tags), corev1.IPv4Protocol, "")
+	createLaunchTemplateInput.DryRun = aws.Bool(true)
+
+	if _, err := n.ec2api.CreateLaunchTemplate(ctx, createLaunchTemplateInput); awserrors.IgnoreDryRunError(err) != nil {
+		nodeClass.StatusConditions().SetFalse(v1.ConditionTypeValidationSucceeded, "LaunchTemplateAuthCheckFailed", fmt.Sprintf("unauthorized operation %s", err))
+		return reconcile.Result{}, reconcile.TerminalError(fmt.Errorf("create launch template %w", err))
+	}
+
+	amis := nodeClass.Status.AMIs
+	if amis == nil {
+		return reconcile.Result{}, fmt.Errorf("failed to list AMIs from NodeClass status, %w", err)
+	}
+
+	var instanceType ec2types.InstanceType
+	if len(nodeClass.Status.AMIs) > 0 {
+		for _, req := range nodeClass.Status.AMIs[0].Requirements {
+			if req.Key == "kubernetes.io/arch" && len(req.Values) > 0 {
+				switch req.Values[0] {
+				case "amd64":
+					instanceType = ec2types.InstanceTypeT3Micro
```

Comment:
Can we pick a more common instance type -- like something in the `m` category? Some customers have SCPs/condition keys that block launching with specific instance types and I'd like to avoid failing validation here if we can due to picking an instance type that wasn't allowed through that policy